packages/frontend-shared/src/composables/examples/UseCohortsExample.cy.tsx

```diff
@@ -0,0 +1,21 @@
+import { WEIGHTED } from '../../utils/weightedChoice'
+import UseCohortsExample, { CopyOption } from './UseCohortsExample.vue'
+
+describe('useCohorts example', () => {
+  const copyOptions: CopyOption[] = [
+    { cohort: 'A', value: 'Notification Title A' },
+    { cohort: 'B', value: 'Notification Title B' },
+  ]
+
+  it('should show value for one cohort with default algorithm', () => {
+    cy.mount(() => <UseCohortsExample copyOptions={copyOptions}/>)
+    cy.findByTestId('result').contains('Notification Title')
+  })
+
+  it('should show value for one cohort with supplied algorithm', () => {
+    const weighted25_75 = WEIGHTED([25, 75])
```

Comment:
I can't see where we actually choose the cohort - is this in another PR somewhere? Eg - what is the ratio between A and B, and how often am I in A vs B?

I wasn't as deeply involved in the finer details of the ACI project, I may have missed something obvious that has already been discussed somewhere.

Reply:
The WeightedAlgorithm is what actually selects the cohort in `useCohorts`.  The `fetchCohort` function in `useCohorts` is where the logic occurs.  The logic checks the `cache` file to see if a cohort has already been selected for a given "name" (this is the term that Mike commented is confusing, which I agree with).  If the cohort for that name has already been selected, then it just returns that value (the "else" case), if it has not been selected, then it uses the configured weighted algorithm to choose one (`algorithm.pick`) and then store it in the cache and return that choice.  

The default algorithm is the `WEIGHTED_EVEN` one which is being used by the Spec list banners to provide even weight to either copy choice being tested in the banners.  I provided a way to customize the weighting after hearing about a potential future case where the A/B testing would be run on a 25/75 weighting.